### PR TITLE
build: College ERP Model

### PR DESCRIPTION
**Pull request by kevit-aayush-popat**
College ERP Model

### CODE REVIEW

**Review by kevit-misri-pandya — APPROVED** [review]

File: src/components/attendance/attendance.types.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { ObjectId } from 'mongoose'
+import { Type } from 'typescript'
+
+export interface attendanceBody {
+    studentOriginalId: ObjectId
+    studentId: string
+    date: string
+    status: any
```
Comment (kevit-misri-pandya) [inline src/components/attendance/attendance.types.d.ts:8]: Don't Use Any! Can delare an enum for PRESENT , ABSENT status

**Review by kevit-misri-pandya — CHANGES_REQUESTED** [review]

**Comment by kevit-aayush-popat** [comment]
changes in attendance module are completed

File: lib/__tests__/attendance.test.js
Comment (kevit-misri-pandya) [inline lib/__tests__/attendance.test.js:1]: @kevit-aayush-popat Add a Lib in .gitignore

File: src/app.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
+// import * as http from 'http'
```
Comment (kevit-misri-pandya) [inline src/app.ts:1]: @kevit-aayush-popat Remove the commented code.

File: src/components/attendance/attendance.controller.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+     * @param {NextFunction} next - The Express NextFunction.
+     * @returns {Promise<void>} - A Promise that resolves with the list of students with low attendance.
+     */
+
+    async lowAttendanceList(
+        req: Request,
+        res: Response,
+        next: NextFunction
+    ): Promise<void> {
+        // console.log(new Date('Wed Oct 19 2023'))
```
Comment (kevit-misri-pandya) [inline src/components/attendance/attendance.controller.ts:79]: Remove console

File: src/components/attendance/attendance.error.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    ATTENDANCE_ERROR: 'No attendnce marked',
+    ATTENDANCE_ID_NOT_FOUND: 'User Id not found',
+    CREATE_ATTENDANCE_UNHANDLED_IN_DB:
+        'Something went wrong while creating new attendance',
+    EMAIL_OR_PASSWORD_NOT_FOUND: 'Email or Password not present.',
+    INVALID_ATTENDANCE_UPDATES: 'Invalid Updates',
+    ATTENDANCE_ALREADY_EXIST: 'ATTENDANCE already exists',
+    NO_ATTENDANCE_FOUND: 'no attendance found',
+    UNABLE_TO_PROCESS_REQUEST:'Unable to request process'
+}
```
Comment (kevit-misri-pandya) [inline src/components/attendance/attendance.error.ts:12]: @kevit-aayush-popat Make Sure Alphabetical Order is Correct in Error Messages check in all Files

File: src/components/students/students.DAL.ts [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+            err
+        )
+    }
+}
+
+export async function checkAttendance(student: {
+    studentId: any
+    studentOriginalId?: mongoose.Types.ObjectId
+    date: any
+    status?: any
```
Comment (kevit-misri-pandya) [inline src/components/students/students.DAL.ts:122]: @kevit-aayush-popat Don't use any

File: src/components/students/students.controller.ts [hunk 6]
```diff
@@ -0,0 +1,209 @@
+/// //left
```
Comment (kevit-misri-pandya) [inline src/components/students/students.controller.ts:1]: ??

**Review by kevit-misri-pandya — CHANGES_REQUESTED** [review]

**Comment by kevit-aayush-popat** [comment]
refactored the changes assigned

**Review by kevit-misri-pandya — APPROVED** [review]

**Pull request merged**